packages/protocol/contracts/tokenvault/erc20/translators/UsdcTranslator.sol

```diff
@@ -27,32 +31,46 @@ contract UsdcTranslator is BaseTranslator {
     /// @dev Only an ERC20Vault can call this function.
     /// @param account The account to mint tokens to.
     /// @param amount The amount of tokens to mint.
-    function mint(address account, uint256 amount) public onlyFromNamed("erc20_vault") {
-        IERC20TokenVaultRelayer(nativeTokenContract).mint(account, amount);
+    function mint(
+        address token,
+        address account,
+        uint256 amount
+    )
+        public
+        onlyFromNamed("erc20_vault")
```

Comment:
I think you can just remove this check, the important things is that msg.sender will be the ERC20Vault for the USDC token contract, so only delegate calls done by ERC20Vault will have that property.